lib/publiccloud/aws_client.pm

```diff
@@ -0,0 +1,110 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Helper class for amazon connection and authentication
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>, qa-c team <qa-c@suse.de>
+
+package publiccloud::aws_client;
+use Mojo::Base -base;
+# use Mojo::JSON 'decode_json';
+use testapi;
+# use mmapi 'get_current_job_id';
+
+use constant CREDENTIALS_FILE => '/root/amazon_credentials';
+
+has key_id => undef;
+has key_secret => undef;
+has security_token => undef;
+has region => undef;
+has ssh_key => undef;
+has ssh_key_file => undef;
+has credentials => undef;
+has vault => undef;
+has aws_account_id => undef;
+has service => undef;
+
+sub vault_create_credentials {
```

Comment:
No for this PR but we are coming closer to a point, where a independent vault class might start to make sense

Reply:
I don't think so , I think it is in right place . because to put this code into vault class you need to make it aware  about what is ECR/EC2/EKS because we have different logic for each so IMO this is right place for this logic no matter that it has `vault_create_credentials` name

Reply:
agreed